streams/src/main/java/org/apache/kafka/streams/state/KeyValueStore.java

```diff
@@ -27,7 +27,7 @@
  * @param <K> The key type
  * @param <V> The value type
  */
-public interface KeyValueStore<K, V> extends StateStore, ReadOnlyKeyValueStore<K, V> {
+public interface KeyValueStore<K, V> extends StateStore, ReadOnlyKeyValueStore<K, V>, PrefixScanKeyValueStore<K, V> {
```

Comment:
Adding this new interface to KeyValueStore may have some serious implications for implementers. Are we sure we have to?

At least, we need to add a default implementation.

Reply:
I am not sure we have to add this to all KeyValueStores. I had previously tried to limit it to the RocksDB instance only, but previously-given feedback suggested that it should support both caching and logging for performance related reasons. What we have now is due partially to a snowballing of inclusion of caching, logging and rocksDB store, and partially due to my thinking that "If Range can work in all tables, why not prefixScan since it's basically the same operation?"

The PrefixScanKeyValueStore needs to support both get() (for LHS->RHS lookup) and prefixScan() (RHS->LHS lookup), and get() is tied to ReadOnlyKeyValueStore. I also didn't want to expose prefixScan to the outside world via ReadOnlyKeyValueStore.

If you (or anyone else) feels strongly about removing it, please direct me on how you would see best to limit it to RocksDB + Caching + Logging. I am not confident in messing with the existing structure too much, so guidance / direction on it would be preferred than leaving it strictly up to me. 👍